src/renderer.ts

```diff
@@ -1,8 +1,28 @@
-import { gameState, ctx, bank } from "./main";
+import { gameState, ctx, bank, canvas, canvas_W, canvas_H } from "./main";
 import { RoomMap } from "./room_map";
 
 export class Renderer {
-	constructor() { }
+	public zoomScale: number;
+	public zoomScaleNext: Map<number, number>;
+
+	constructor() {
+		this.zoomScale = 1;
+		this.zoomScaleNext = new Map<number, number>([[1, 1.25], [1.25, 1.5], [1.5, 1.75], [1.75, 2], [2, 1]]);
+	}
+
+	public disableSmoothing(): void {
+		ctx.webkitImageSmoothingEnabled = false;
+		ctx.mozImageSmoothingEnabled = false;
+		ctx.imageSmoothingEnabled = false;
+	}
+
+	public scale(zoomScale?: number): void {
+		this.zoomScale = zoomScale == null ? this.zoomScaleNext.get(this.zoomScale) : zoomScale;
```

Comment:
Si `zoomScale` n'est pas null et que `zoomScaleNext` ne contient pas la clé `zoomScale`, `this.zoomScale` peut être null